### DISCUSSION TITLE
Feature Request: File Uploads in Prompts

### DISCUSSION THREAD

**Discussion by fredparadis** [microsoft/sample-app-aoai-chatGPT#1057] [Ideas]
I would like to enhance the user experience by allowing users to upload files (e.g., images, documents) as part of their prompts. This would enable richer interactions and more comprehensive queries.  
  
Integrate a file upload feature in the chat interface that allows users to attach files. The backend should process these files and include the relevant data in the prompt sent to Azure OpenAI.  
  
Implementing a custom solution using `multer` for file uploads and processing files before sending them to Azure OpenAI.  
  
This feature is available in the Chat Playground, and it would be highly beneficial to have similar functionality in this sample app.  
  
**Example UI/UX**  
- An `<input type="file" multiple>` element in the chat interface.  
- Backend handling of file uploads using middleware like `multer`.  
- Pre-processing of files (e.g., OCR for images) before including the data in prompts.  
  
Thank you for considering this feature request!

**Comment by stbere** [comment]
I would love a feature like this. We could even use the existing cosmosdb storage to temporarily store the attached file to run the AI Search against the documents, then the file could be automatically deleted from the storage after the user ends the chat session.
